galley/pkg/config/analysis/analyzer.go

```diff
@@ -14,41 +14,89 @@
 
 package analysis
 
-import "istio.io/istio/galley/pkg/config/scope"
+import (
+	"istio.io/istio/galley/pkg/config/collection"
+	"istio.io/istio/galley/pkg/config/scope"
+)
 
 // Analyzer is an interface for analyzing configuration.
 type Analyzer interface {
-	Name() string
+	Metadata() Metadata
 	Analyze(c Context)
 }
 
+// Metadata represents metadata for an analyzer
+type Metadata struct {
```

Comment:
Nit: consider moving this to its own file.

Reply:
OK.